lib/extract-stream.js

```diff
@@ -9,28 +9,33 @@ function extractStream (dest, opts) {
   const sawIgnores = new Set()
   return tar.x({
     cwd: dest,
-    filter: (name, entry) => !entry.header.type.match(/^.*link$/i),
+    // Use endsWith rather than match with regex. Should be faster.
+    filter: (name, entry) => !entry.header.type.endsWith('link'),
     strip: 1,
     onwarn: msg => opts.log && opts.log.warn('tar', msg),
     uid: opts.uid,
     gid: opts.gid,
     onentry (entry) {
-      if (entry.type.toLowerCase() === 'file') {
+      // Only lowerCase once.
+      const entryType = entry.type.toLowerCase()
+      if (entryType === 'file') {
         entry.mode = opts.fmode & ~(opts.umask || 0)
-      } else if (entry.type.toLowerCase() === 'directory') {
+      } else if (entryType === 'directory') {
```

Comment:
This was this way because either node-tar or tar-fs have the type capitalized. idr which one does which anymore.